macosx/linux build error

The current build action failed here:

/Users/runner/work/osgearth/osgearth/src/osgEarth/rtree.h:103:35: error: non-local lambda expression cannot have a capture-default
        CALLBACK_TYPE callback = [&](const DATATYPE&) { return true; }) const;
                                  ^
45 warnings and 1 error generated.

--- a/src/osgEarth/TileMesher.cpp
+++ b/src/osgEarth/TileMesher.cpp
@@ -595,7 +595,7 @@ TileMesher::createMeshWithConstraints(
 
                         // Note: the part was already transformed in a previous step.
 
-                        auto& bb = part->getBounds();
+                        const auto& bb = part->getBounds();
                         if (part->isPolygon() && bb.intersects(localBounds))
                         {
                             if (edit.removeExterior)

--- a/src/osgEarth/rtree.h
+++ b/src/osgEarth/rtree.h
@@ -100,7 +100,7 @@ public:
     using DefaultCallbackType = std::function<bool(const DATATYPE&)>;
     template<typename CALLBACK_TYPE = DefaultCallbackType>
     int Search(const ELEMTYPE a_min[NUMDIMS], const ELEMTYPE a_max[NUMDIMS],
-        CALLBACK_TYPE callback = [&](const DATATYPE&) { return true; }) const;
+        CALLBACK_TYPE callback = [](const DATATYPE&) { return true; }) const;
 
     /// Remove all entries from tree
     void RemoveAll();

